src/full/Agda/TypeChecking/MetaVars/Occurs.hs

```diff
@@ -242,6 +243,7 @@ metaCheck m = do
       , text . show $ getQuantity mmod
       ]
     allowAssign <- asksTC envAssignMetas
+    firstOrder <- optFirstOrder <$> pragmaOptions
```

Comment:
Rather than binding `allowAssign` and `firstOrder` here, use `unlessM` and `whenM` below.

Reply:
In this case I disagree that using `unlessM` or `whenM` is prettier. Also the `allowAssign` was already present, I just followed the same style.

Reply:
Maybe not prettier, but saves cycles.  Using `unlessM` and `whenM` you get the laziness that is usual for Haskell, with bind you are unnecessarily strict.